validator/remote/src/integration-test/java/tech/pegasys/teku/validator/remote/typedef/OkHttpValidatorTypeDefClientTest.java

```diff
@@ -0,0 +1,161 @@
+
+  private final MockWebServer mockWebServer = new MockWebServer();
+  private final OkHttpClient okHttpClient = new OkHttpClient.Builder().build();
+
+  @BeforeEach
+  public void beforeEach(SpecContext specContext) throws Exception {
+    mockWebServer.start();
+    okHttpValidatorTypeDefClient =
+        new OkHttpValidatorTypeDefClient(
+            okHttpClient, mockWebServer.url("/"), specContext.getSpec(), false);
+    dataStructureUtil = specContext.getDataStructureUtil();
+  }
+
+  @AfterEach
+  public void afterEach() throws Exception {
+    mockWebServer.shutdown();
+  }
+
+  @TestTemplate
+  void registerValidators_makesJsonRequest() throws InterruptedException, JsonProcessingException {
+
+    mockWebServer.enqueue(new MockResponse().setResponseCode(200));
+
+    SszList<SignedValidatorRegistration> validatorRegistrations =
+        dataStructureUtil.randomValidatorRegistrations(5);
+
+    String expectedRequest =
+        JsonUtil.serialize(
+            validatorRegistrations,
+            ApiSchemas.SIGNED_VALIDATOR_REGISTRATIONS_SCHEMA.getJsonTypeDefinition());
+
+    okHttpValidatorTypeDefClient.registerValidators(validatorRegistrations);
+
+    RecordedRequest recordedRequest = mockWebServer.takeRequest();
+
+    assertThat(recordedRequest.getPath()).isEqualTo("/eth/v1/validator/register_validator");
+    assertThat(recordedRequest.getMethod()).isEqualTo("POST");
+    assertThat(recordedRequest.getHeader("Content-Type"))
+        .isEqualTo("application/json; charset=utf-8");
+
+    String actualRequest = recordedRequest.getBody().readUtf8();
+
+    assertJsonEquals(actualRequest, expectedRequest);
+  }
+
+  @TestTemplate
+  void registerValidators_handlesFailures() {
+    mockWebServer.enqueue(
+        new MockResponse().setResponseCode(400).setBody("{\"code\":400,\"message\":\"oopsy\"}"));
+
+    SszList<SignedValidatorRegistration> validatorRegistrations =
+        dataStructureUtil.randomValidatorRegistrations(5);
+
+    IllegalArgumentException badRequestException =
+        Assertions.assertThrows(
+            IllegalArgumentException.class,
+            () -> okHttpValidatorTypeDefClient.registerValidators(validatorRegistrations));
+
+    assertThat(badRequestException.getMessage())
+        .contains("Invalid params response from Beacon Node API");
+
+    mockWebServer.enqueue(
+        new MockResponse()
+            .setResponseCode(500)
+            .setBody("{\"code\":500,\"message\":\"Internal server error\"}"));
+
+    RemoteServiceNotAvailableException serverException =
+        Assertions.assertThrows(
+            RemoteServiceNotAvailableException.class,
+            () -> okHttpValidatorTypeDefClient.registerValidators(validatorRegistrations));
+
+    assertThat(serverException.getMessage()).contains("Server error from Beacon Node API");
+  }
+
+  @TestTemplate
+  void registerValidators_makesSszRequestIfSszEncodingPreferred() throws InterruptedException {
```

Comment:
probably could have a similar test that server returns a 415 and see that we end up with json returned...

so we send octet, server returns 415, we send json. (and we dont need to decode json again, just need to see the content-type is json and that the actual second call happens... then if we want to extend it a tiny bit, make a 'second' request, and that just goes straight to json...